modules/300-prometheus/templates/prometheus/longterm/prometheus.yaml

```diff
@@ -133,8 +133,30 @@ spec:
       checksum/kube-rbac-proxy: {{ include "helm_lib_kube_rbac_proxy_ca_certificate" (list . "") | sha256sum }}
   secrets:
   - prometheus-api-client-tls
-  {{- include "helm_lib_node_selector" (tuple . "monitoring") | nindent 2 }}
-  {{- include "helm_lib_tolerations" (tuple . "monitoring") | nindent 2 }}
+  {{- if .Values.prometheus.longtermNodeSelector }}
+  nodeSelector: {{ .Values.prometheus.longtermNodeSelector | toJson }}
+  {{- else if gt (index .Values.global.discovery.d8SpecificNodeCountByRole "prometheus-longterm" | int) 0 }}
```

Comment:
Please, remove this else, it was a bad decision to use nore-role here. It won't fit the [allowed list of labels](https://github.com/deckhouse/deckhouse/blob/main/modules/340-monitoring-custom/hooks/reserved_domain_nodes.go).

Reply:
Ok, got it. Back to the original approach for explicit setup to support backward compatibility